aya/src/interface.rs

```diff
@@ -0,0 +1,168 @@
+//! Network interfaces representing kernel network interfaces.
+
+use std::ffi::CStr;
+
+use libc::if_nameindex;
+
+#[derive(Debug, Clone)]
+/// A kernel network interface.
+// if_name isn't stored because it can change in the kernel and aya won't know
+pub struct NetworkInterface {
+    pub(crate) index: i32,
+}
+
+impl NetworkInterface {
+    /// Provides a number that can be used to identify this interface on this system.
+    pub fn index(&self) -> i32 {
+        self.index
+    }
+
+    /// Extracts the interface name from the kernel.
+    pub fn name(&self) -> Result<String, std::io::Error> {
+        let mut buffer: [libc::c_char; libc::IF_NAMESIZE] = [0; libc::IF_NAMESIZE];
+        let name = unsafe {
+            // Returns null on error
+            let res = libc::if_indextoname(self.index as u32, buffer.as_mut_ptr());
+
+            if res.is_null() {
+                return Err(std::io::Error::last_os_error());
+            }
+
+            CStr::from_ptr(buffer.as_ptr())
+        };
+
+        Ok(name.to_string_lossy().to_string())
```

Comment:
I wasn't really sure how to do this but I eventually got this version working as expected.